src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -373,36 +377,29 @@ private Vector limitDirectionSlewRate(Vector spdVec) {
     double angleDif =
         SwerveUtils.AngleDifference(inputTranslationDir, m_currentTranslationDirRadians);
 
-    m_prevSlewRateTime = currentTime;
-
     if (angleDif < DriveConfig.MIN_ANGLE_SLEW_RATE) {
       m_currentTranslationDirRadians =
           SwerveUtils.StepTowardsCircular(
               m_currentTranslationDirRadians, inputTranslationDir, directionSlewRate * elapsedTime);
       m_currentTranslationMag = m_magLimiter.calculate(inputTranslationMag);
-      return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
-    }
-
-    if (angleDif > DriveConfig.MAX_ANGLE_SLEW_RATE) {
-      if (SwerveUtils.approxEqual(
-          m_currentTranslationMag,
-          0)) { // some small number to avoid floating-point errors with equality checking
-        // keep currentTranslationDir unchanged
+      SmartDashboard.putNumber("translation magnitude output", inputTranslationMag);
+    } else if (angleDif > DriveConfig.MAX_ANGLE_SLEW_RATE) {
+      if (m_currentTranslationMag > 1e-4) { // some small number to avoid floating-point errors with equality checking
         m_currentTranslationMag = m_magLimiter.calculate(0.0);
-        return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
+      } else {
+        m_currentTranslationDirRadians =
+            SwerveUtils.WrapAngle(m_currentTranslationDirRadians + Math.PI);
+        m_currentTranslationMag = m_magLimiter.calculate(inputTranslationMag);
       }
-
+    } else {
       m_currentTranslationDirRadians =
-          SwerveUtils.WrapAngle(m_currentTranslationDirRadians + Math.PI);
-      m_currentTranslationMag = m_magLimiter.calculate(inputTranslationMag);
-      return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
-    }
+          SwerveUtils.StepTowardsCircular(
+              m_currentTranslationDirRadians, inputTranslationDir, directionSlewRate * elapsedTime);
 
-    m_currentTranslationDirRadians =
-        SwerveUtils.StepTowardsCircular(
-            m_currentTranslationDirRadians, inputTranslationDir, directionSlewRate * elapsedTime);
+      m_currentTranslationMag = m_magLimiter.calculate(0.0);
 
-    m_currentTranslationMag = m_magLimiter.calculate(0.0);
+      m_prevSlewRateTime = currentTime;
+    }
 
     return (new Vector(m_currentTranslationMag, 0)).rot(m_currentTranslationDirRadians);
```

Comment:
Do not allocate new objects. return a modified existing vector. See the drivetrain memory fixes for inspiration.

Reply:
modified spdVec

Reply:
If you're modifying it do you need to return it anymore?

Reply:
method is void now and spdCommanded is gone, so fewer vectors copied and better RAM usage!